spark_log_parser/parsing_models/stage_model.py

```diff
@@ -7,8 +7,21 @@ class StageModel:
     Model for a stage within a spark job
     """
     def __init__(self):
-        self.start_time = -1
-        self.tasks = []
+        self.id: int = None
+        self.stage_name: str = None
+        # By default, Stages will have a "Stage Attempt ID" of 0 the first time they are run.
+        # If a Stage fails for some reason, it may be re-tried, and appear again in the eventlog with the
+        # "Stage Attempt ID" incremented by 1. As such, we want to keep track of this so that we aren't putting
+        # "stale" data onto a Stage when processing the eventlog lines (i.e. we don't want to set the `submission_time`
+        # to be that of Attempt ID 0 when we've already encountered Attempt ID 1!)
+        self.attempt_id: int = None
+        self.stage_info: dict = None
+        self.start_time: int = -1
+        self.tasks: list[TaskModel] = []
+        self.tasks_finalized: bool = False
+        self.num_tasks: int = 0
+        self.submission_time: int = None
+        self.completion_time: int = None
```

Comment:
These fields were being populated here - https://github.com/synccomputingcode/spark_log_parser/pull/27/files#diff-e2e3163cf71093ffe154e86a1333f2db49453440abcb815275c8d49c04a584f3L156-L165 - and here - https://github.com/synccomputingcode/spark_log_parser/pull/27/files#diff-e2e3163cf71093ffe154e86a1333f2db49453440abcb815275c8d49c04a584f3L174-L176 - so I just formalized this and put them onto the `StageModel` directly.